cnd/src/swap_protocols/asset.rs

```diff
@@ -21,22 +21,40 @@ pub trait Asset:
     + FromHttpAsset
     + Into<AssetKind>
 {
-    fn equal_or_greater_value(&self, other: &Self) -> bool;
+    fn compare_to(&self, other: &Self) -> i8;
 }
 
 impl Asset for BitcoinQuantity {
-    fn equal_or_greater_value(&self, other: &BitcoinQuantity) -> bool {
-        self >= other
+    fn compare_to(&self, other: &BitcoinQuantity) -> i8 {
```

Comment:
📣 Use an enum (or is it not possible?)

Reply:
Sorry, I don't understand. can you elaborate on that please?

Reply:
```
enum Compare {
  GreaterThan,
  LesserThan,
  Equal
}
```

return `Compare` instead of an `i8`.